src/plugins/data/public/query/query_string/query_history.ts

```diff
@@ -4,60 +4,82 @@
  */
 
 import { BehaviorSubject } from 'rxjs';
-import { DataStorage, Dataset } from '../../../common';
+import { DataStorage } from '../../../common';
 import { Query, TimeRange } from '../..';
 
-// Todo: Implement a more advanced QueryHistory class when needed for recent query history
+const MAX_HISTORY_SIZE = 500;
+export const HISTORY_KEY_PREFIX = 'query_';
+
 export class QueryHistory {
-  constructor(private readonly storage: DataStorage) {}
+  private changeEmitter: BehaviorSubject<any[]>;
 
-  private changeEmitter = new BehaviorSubject<any[]>(this.getHistory() || []);
+  constructor(private readonly storage: DataStorage) {
+    this.changeEmitter = new BehaviorSubject<any[]>(this.getHistory());
+  }
 
-  getHistoryKeys() {
+  public getHistoryKeys(): string[] {
     return this.storage
       .keys()
-      .filter((key: string) => key.indexOf('query_') === 0)
-      .sort()
-      .reverse();
+      .filter((key: string) => key.startsWith(HISTORY_KEY_PREFIX))
+      .sort((a, b) => {
+        const timeA = parseInt(a.split('_')[1], 10);
+        const timeB = parseInt(b.split('_')[1], 10);
+        return timeB - timeA; // Sort in descending order (most recent first)
+      });
   }
 
-  getHistory() {
-    return this.getHistoryKeys().map((key) => this.storage.get(key));
+  public getHistory(): any[] {
+    return this.getHistoryKeys()
+      .map((key) => this.storage.get(key))
+      .sort((a, b) => b.time - a.time);
```

Comment:
getHistoryKeys already sorts by time, is this sort needed?

Reply:
Good catch. I added that sort afterwards. This is no longer needed